File: aspnet-core/GBS/ABS.Logic.Shared/AirbrakeNotifier.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Configuration;

namespace ABS.Logic.Shared.Airbrake
{
	public class AirbrakeNotifier
	{
		public AirbrakeNotifier(AirbrakeConfig config, string logger = null, string httpRequestHandler = null)
		{

		}

		public void AddFilter(Func<Notice, Notice> filter)
		{

		}
		public void Notify(Exception exception, IHttpContext context = null, Severity severity = null)
		{

		}
		public void NotifyAsync(Exception exception, IHttpContext context = null, Severity severity = null)
		{

		}

		//public AirbrakeConfig New ()
		//{
		//	return null;
		//}
		//public Exception Notify ()
		//{
		//	return null;
		//}
		//public Exception NotifyAsync (AirbrakeConfig config)
		//{
		//	return null;
		//}
	}

	public class Severity
	{
	}

	public interface IHttpContext
	{
	}

	public class Notice
	{
	}

	public class AirbrakeConfig
	{
		public string Environment { get; set; }
		public string AppVersion { get; set; }
		public string ProjectKey { get; set; }
		public string ProjectId { get; set; }
		public string Host { get; set; }
		public string LogFile { get; set; }
		public string ProxyUri { get; set; }
		public string ProxyUsername { get; set; }
		public string ProxyPassword { get; set; }
		public IList<string> IgnoreEnvironments { get; set; }
		public IList<string> WhitelistKeys { get; set; }
		public IList<string> BlacklistKeys { get; set; }

		//public string Environment = string.Empty;
		//public string AppVersion = string.Empty;
		//public string ProjectKey = string.Empty;
		//public string ProjectId = string.Empty;
		//public string Host = string.Empty;
		//public string LogFile = string.Empty;
		//public string ProxyUri = string.Empty;
		//public string ProxyUsername = string.Empty;
		//public string ProxyPassword = string.Empty;
		//public IList<string> IgnoreEnvironments = null;
		//public IList<string> WhitelistKeys = null;
		//public IList<string> BlacklistKeys = null;

		//public string New ()
		//{
		//	return null;
		//}

		public static AirbrakeConfig Load(IDictionary<string, string> settings)
		{
			return new AirbrakeConfig();
		}
	}
}

File: Server/aspnet-core/eSWIS/eSWIS.Shared/AirbrakeNotifier.vb
Imports System

'Namespace Sharpbrake.Client
Public Class AirbrakeNotifier
        'Public Sub New(config As AirbrakeConfig, Optional logger As ILogger = Nothing, Optional httpRequestHandler As IHttpRequestHandler = Nothing)

        'End Sub

        'Public Event NotifyCompleted As NotifyCompletedEventHandler

        'Public Sub AddFilter(filter As Func(Of Notice, Notice))

        'End Sub
        'Public Sub Notify(exception As Exception, Optional context As IHttpContext = Nothing, Optional severity As Severity = Severity.Error)

        'End Sub
        'Public Sub NotifyAsync(exception As Exception, Optional context As IHttpContext = Nothing, Optional severity As Severity = Severity.Error)

        'End Sub
        'Protected Overridable Sub OnNotifyCompleted(eventArgs As NotifyCompletedEventArgs)

        'End Sub
        Public Sub New(config As AirbrakeConfig)

        End Sub

        'Public Event NotifyCompleted As NotifyCompletedEventHandler

        'Public Sub AddFilter(filter As Func(Of Notice, Notice))

        'End Sub
        Public Sub Notify(exception As Exception)

        End Sub
        Public Sub NotifyAsync(exception As Exception)

        End Sub
        'Protected Overridable Sub OnNotifyCompleted(eventArgs As NotifyCompletedEventArgs)

        'End Sub
    End Class

    Public Class AirbrakeConfig
    Public Sub New()

    End Sub

    Public Property Environment As String
        Public Property AppVersion As String
        Public Property ProjectKey As String
        Public Property ProjectId As String
        Public Property Host As String
        Public Property LogFile As String
        Public Property ProxyUri As String
        Public Property ProxyUsername As String
        Public Property ProxyPassword As String
        Public Property IgnoreEnvironments As IList(Of String)
        Public Property WhitelistKeys As IList(Of String)
        Public Property BlacklistKeys As IList(Of String)

        Public Shared Function Load(settings As IDictionary(Of String, String)) As AirbrakeConfig
            Return New AirbrakeConfig()
        End Function
    End Class

'Public Class NotifyCompletedEventArgs
'    Inherits AsyncCompletedEventArgs

'    Public Sub New(response As AirbrakeResponse, [error] As Exception, cancelled As Boolean, state As Object)

'    Public ReadOnly Property Result As AirbrakeResponse
'End Class
'End Namespace
